pre_commit/commands/migrate_config.py

```diff
@@ -21,8 +21,10 @@ def _migrate_map(contents):
     # Find the first non-header line
     lines = contents.splitlines(True)
     i = 0
-    while _is_header_line(lines[i]):
-        i += 1
+    # Only loop on non empty configuration file
+    if i < len(lines):
+        while _is_header_line(lines[i]):
```

Comment:
I *think* this can be `while i < len(lines) and _is_header_line(...)` -- that way we don't overrun a file that's just `\n\n` as well

Reply:
Thanks for the tips. I did consider adding a test but I wasn't sure something so small needed it. People weren't kidding when they said working with code written by someone else isn't easy.

Reply:
heh, it is a pretty small change but I want to make sure we don't regress int he future 😆

Reply:
Yeah I get that. Thanks for guiding me through it and putting up with me being slow. I ran the test and it passed so it seems to be all good this time. You were spot on about my original fix not catering to new lines.

Reply:
that's why we test -- looks like it's uhh, failing `pre-commit` checks now ;)

Reply:
Yeah I saw. I feel so bad for wasting your time like this. You can just close the pull request and forget it if you like. As it turns out I'm more clueless working on this than I thought I would be.

Reply:
no! it's absolutely fine -- no urgency whatsoever <3 I'm happy to help you through the process!

If you click through the travis-ci output it autofixed it for you -- you just need to apply the same change to the branch :)

I think it just needs one extra newline right above the test you added

Reply:
Ah yeah, that'd be a pep8 rule. Makes the code nicely spaced I guess

Reply:
Well that was a learning experience. Thanks again for taking the time to talk me through it. Hopefully next time I try I'm a little less rusty.

Reply:
No problem at all! I'm anyways happy to help someone fix things :)